DuckDuckGo/NetworkProtectionVisibilityForTunnelProvider.swift

```diff
@@ -0,0 +1,52 @@
+//
+//  NetworkProtectionVisibilityForTunnelProvider.swift
+//  DuckDuckGo
+//
+//  Copyright © 2024 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+#if NETWORK_PROTECTION
+
+import Foundation
+
+struct NetworkProtectionVisibilityForTunnelProvider: NetworkProtectionFeatureVisibility {
+    func isWaitlistBetaActive() -> Bool {
+        preconditionFailure("Does not apply to Tunnel Provider")
+    }
+
+    func isWaitlistUser() -> Bool {
+        preconditionFailure("Does not apply to Tunnel Provider")
+    }
+    
+    // todo - https://app.asana.com/0/0/1206844038943626/f
+    func isPrivacyProLaunched() -> Bool {
+#if SUBSCRIPTION && ALPHA
+        true
+#else
+        false
+#endif
+    }
+    
+    // todo - https://app.asana.com/0/0/1206844038943626/f
+    func shouldMonitorEntitlement() -> Bool {
+#if SUBSCRIPTION && ALPHA
+        true
+#else
+        isPrivacyProLaunched()
+#endif
+    }
```

Comment:
Still need to figure out how these should be implemented in the tunnel provider as we don't have access to Feature Flagger there.

Reply:
The way this works on macOS is the entitlement monitoring is always started if the packet tunnel provider was launched with a subscription access token. @graeme was changing that logic earlier today though so we can confirm it on Monday.

Reply:
Smart move 👍 I'd probably do the same.